app/questionnaire/router.py

```diff
@@ -129,18 +128,18 @@ def get_next_location_url(
             ):
                 return return_to_url
 
-            if is_last_block_in_section:
-                return self._get_next_location_url_for_last_block_in_section(location)
+            return self._get_next_location_url_for_complete_section(location)
 
         # Due to backwards routing, you can be on the last block without the section being complete
+        is_last_block_in_section = routing_path[-1] == location.block_id
```

Comment:
This could be removed, wasn't convinced routing_path[-1] == location.block_id was obvious what it was doing, the var helps.